drivers/lora/sx126x.c

```diff
@@ -410,7 +410,10 @@ static void sx126x_dio1_irq_work_handler(struct k_work *work)
 		Radio.IrqProcess();
 	}
 
-	sx126x_dio1_irq_enable(&dev_data);
+	/* Re-enable the interrupt if we are not in sleep mode */
+	if (dev_data.mode != MODE_SLEEP) {
```

Comment:
Is the radio going to sleep state in the middle of the handler? Did you hit a scenario?

Reply:
The issue mentioned in the comment says "...DIO1 interrupt line is left enabled. This results in additional power consumption on architectures where turning on interrupts is not free from a power perspective."

Guess it makes sense, what's the specific architecture scenario/setup? Don't think I've hit such a thing...

Reply:
> The issue mentioned in the comment says "...DIO1 interrupt line is left enabled. This results in additional power consumption on architectures where turning on interrupts is not free from a power perspective."

I agree that turning ON the IRQ always in the IRQ handler for standalone radios is a waste of resources but I'm confused about the radio being in SLEEP state.

Reply:
> Is the radio going to sleep state in the middle of the handler? Did you hit a scenario?

`Radio.IrqProcess();` is the place where the radio transitions to sleep mode, through
https://github.com/zephyrproject-rtos/zephyr/blob/770732fe92faab44d93bcbc00fc4d45476c24dea/drivers/lora/sx12xx_common.c#L101
https://github.com/zephyrproject-rtos/zephyr/blob/770732fe92faab44d93bcbc00fc4d45476c24dea/drivers/lora/sx126x.c#L286-L290

Unconditionally turning on the interrupt after those handlers run is just overriding the `MODE_SLEEP` logic. This can be easily seen by putting a `printk` inside `sx126x_dio1_irq_enable` and `sx126x_dio1_irq_disable` and running the `samples/net/lora/send` sample.

Reply:
> Guess it makes sense, what's the specific architecture scenario/setup? Don't think I've hit such a thing...

nRF52 microcontrollers pay a penalty of ~20uA while any pin is configured as an edge-triggered interrupt (Requires the `GPIOTE` peripheral to be enabled)

Reply:
Got it, so this makes a difference because it's the only GPIO interrupt in the whole application, hence shutting it off allows cutting power to the whole `GPIOTE` device. That's a good point, could not wrap my head around power consumption of individual lines, but this makes sense now, thanks for explaining.